### PR TITLE
Migrate Graql Match/Get tests to BDD

### PR DESCRIPTION
**Pull request by alexjpwalker**
## What is the goal of this PR?

To migrate our Graql Match/Get tests to BDD.

This PR forms part of the work towards https://github.com/graknlabs/grakn/issues/5719 .

## What are the changes implemented in this PR?

Adds new BDD tests, many of which were migrated from GraqlGetIT in Grakn Core.

### CODE REVIEW

File: behaviour/graql/language/get.feature [hunk 1]
```diff
@@ -16,8 +16,1007 @@
...
+  Scenario Outline: an error is thrown when getting the `<agg_type>` of an undefined variable in an aggregate query
+    Then graql get throws
+      """
+      match
+        $x isa person;
+      get;
+      <agg_type> $y;
+      """
+
+    Examples:
```
Comment (flyingsilverfin) [inline behaviour/graql/language/get.feature:696]: also great!

File: behaviour/graql/language/get.feature [hunk 2]
```diff
@@ -16,8 +16,1007 @@
...
+    And group identifiers are
+      |      | owner |
+      | gAPP | APP   |
+      | gGOO | GOO   |
+    Then group aggregate values are
+      | group | value |
+      | gAPP  | 4     |
+      | gGOO  | 2     |
 
-  Scenario: Disjunctions return the union of composing query statements
```
Comment (flyingsilverfin) [inline behaviour/graql/language/get.feature:21]: where did this go?

File: behaviour/graql/language/match.feature [hunk 3]
```diff
@@ -191,68 +183,897 @@ Feature: Graql Match Clause
...
+      |     | check | value |
+      | AG9 | key   | ref:0 |
+      | SS9 | key   | ref:1 |
+    Then uniquely identify answer concepts
+      | x   |
+      | AG9 |
+      | SS9 |
+
+
+  Scenario: when an attribute instance is fully specified, `has` matches its owners
```
Comment (flyingsilverfin) [inline behaviour/graql/language/match.feature:831]: can we extend this to match owners from more than 1 type? Eg. add a `Dog` that has a name unrelated to `person` and make sure both dogs and people come back when querying for things with names.

Might even be good to have this working for the other types also owning an attribute: relation and attribute

File: behaviour/graql/language/match.feature [hunk 4]
```diff
@@ -191,68 +183,897 @@ Feature: Graql Match Clause
...
+      |     | check | value |
+      | JH1 | key   | ref:0 |
+      | JH2 | key   | ref:1 |
+    Then uniquely identify answer concepts
+      | x   |
+      | JH1 |
+      | JH2 |
+
+
+  Scenario: `has` with an instance fully specified matches all its owners, even if they own other instances of the same attribute
```
Comment (flyingsilverfin) [inline behaviour/graql/language/match.feature:855]: shorten a bit

File: behaviour/graql/language/match.feature [hunk 5]
```diff
@@ -191,68 +183,897 @@ Feature: Graql Match Clause
...
+    Then uniquely identify answer concepts
+      | x   |
+      | PER |
+
+
+  ##############################
+  # ATTRIBUTE VALUE COMPARISON #
+  ##############################
+
+  Scenario: when two things each own an attribute with the same value, and those attributes have different types, they will match by equality, but not by ownership
```
Comment (flyingsilverfin) [inline behaviour/graql/language/match.feature:885]: same here

File: behaviour/graql/language/match.feature [hunk 6]
```diff
@@ -191,68 +183,897 @@ Feature: Graql Match Clause
...
+      |     | check | value |
+      | DON | key   | ref:1 |
+      | SUS | key   | ref:0 |
+    Then uniquely identify answer concepts
+      | x   |
+      | DON |
+      | SUS |
+
+
+  Scenario: `has $attr > $x` matches owners of any instance `$y` of `$attr` where `$y > $x` even if they also own instance `$z` where `$z < $x`
```
Comment (flyingsilverfin) [inline behaviour/graql/language/match.feature:985]: probably like `attribute value comparison returns a matching subset of all attributes owned by a thing`

File: behaviour/graql/language/match.feature [hunk 7]
```diff
@@ -191,68 +183,897 @@ Feature: Graql Match Clause
...
+      """
+    And concept identifiers are
+      |     | check | value |
+      | JEF | key   | ref:0 |
+    Then uniquely identify answer concepts
+      | x   |
+      | JEF |
+
+
+  Scenario: a negation does not match if the negated block has any matches
```
Comment (flyingsilverfin) [inline behaviour/graql/language/match.feature:1061]: we can talk more about negation i think with @jmsfltchr  and what we want to do to expand these cases!

Comment (flyingsilverfin) [reply behaviour/graql/language/match.feature:1061]: Don't fret about it now though, these are fine :)

File: behaviour/graql/language/get.feature (hunk 2 above)
Comment (alexjpwalker) [reply behaviour/graql/language/get.feature:21]: It emigrated to `match.feature`!

File: behaviour/graql/language/match.feature (hunk 3 above)
Comment (alexjpwalker) [reply behaviour/graql/language/match.feature:831]: Changed insert query to:
```
insert
      $x isa person, has name "Zoe", has age 21, has graduation-date 2020-06-01, has ref 0;
      $y (friend: $x) isa friendship, has age 21, has ref 1;
      $z 2020-06-01 isa graduation-date, has age 21, has ref 2;
      $w isa person, has ref 3;
      $v (friend: $x, friend: $w) isa friendship, has age 7, has ref 4;
      $u 2019-06-03 isa graduation-date, has age 22, has ref 5;
```

File: behaviour/graql/language/match.feature (hunk 4 above)
Comment (alexjpwalker) [reply behaviour/graql/language/match.feature:855]: Changed to
```
`has` matches an attribute's owner even if it owns more attributes of the same type
```

File: behaviour/graql/language/match.feature (hunk 5 above)
Comment (alexjpwalker) [reply behaviour/graql/language/match.feature:885]: `when things own attributes of different types but the same value, they match by equality, but not ownership`

File: behaviour/graql/language/match.feature (hunk 6 above)
Comment (alexjpwalker) [reply behaviour/graql/language/match.feature:985]: How about `when a thing owns multiple attributes of the same type, a value comparison matches if any value matches`? @flyingsilverfin

**Pull request merged**